src/core/TimePicker/TimePicker.tsx

```diff
@@ -0,0 +1,401 @@
+   * Callback when time is changed.
+   */
+  onChange?: (date: Date) => void;
+  /**
+   * Format of the time. 12h or 24h.
+   * @default false
+   */
+  use12Hours?: boolean;
+  /**
+   * Precision of the time.
+   * @default 'minutes'
+   */
+  precision?: 'hours' | 'minutes' | 'seconds';
+  /**
+   * Change step of the hours displayed.
+   * @default 1
+   */
+  hourStep?: number;
+  /**
+   * Change step of the minutes displayed.
+   * @default 1
+   */
+  minuteStep?: number;
+  /**
+   * Change step of the seconds displayed.
+   * @default 1
+   */
+  secondStep?: number;
+};
+
+/**
+ * Time picker component
+ * @example
+ * <TimePicker date={new Date()} onChange={(e) => console.log('New time value: ' + e)} />
+ */
+export const TimePicker = (props: TimePickerProps): JSX.Element => {
+  const {
+    date,
+    onChange,
+    use12Hours = false,
+    precision = 'minutes',
+    hourStep = 1,
+    minuteStep = 1,
+    secondStep = 1,
+  } = props;
+
+  useTheme();
+
+  const [selectedTime, setSelectedTime] = React.useState(date);
+  const [focusedTime, setFocusedTime] = React.useState(
+    selectedTime ?? new Date(),
+  );
+  const [period, setPeriod] = React.useState<'PM' | 'AM' | undefined>(
+    use12Hours ? (focusedTime?.getHours() > 11 ? 'PM' : 'AM') : undefined,
+  );
+
+  React.useEffect(() => {
+    setFocusedTime(date ?? new Date());
+    setSelectedTime(date);
+  }, [date]);
+
+  const onHourClick = (hour: number) => {
+    const adjustedHour = use12Hours ? formatHourFrom12(hour, period) : hour;
+    const adjustedSelectedTime = setHours(
+      adjustedHour,
+      selectedTime ?? new Date(),
+    );
+    updateCurrentTime(adjustedSelectedTime);
+  };
+
+  const onMinutesClick = (minutes: number) => {
+    const adjustedSelectedTime = setMinutes(
+      minutes,
+      selectedTime ?? new Date(),
+    );
+    updateCurrentTime(adjustedSelectedTime);
+  };
+
+  const onSecondsClick = (seconds: number) => {
+    const adjustedSelectedTime = setSeconds(
+      seconds,
+      selectedTime ?? new Date(),
+    );
+    updateCurrentTime(adjustedSelectedTime);
+  };
+
+  const onPeriodClick = (value: string) => {
+    let adjustedSelectedTime = selectedTime ?? new Date();
+    const currentHours = adjustedSelectedTime.getHours();
+    if (value === 'AM' && currentHours > 11) {
+      setPeriod(value);
+      adjustedSelectedTime = setHours(currentHours - 12, adjustedSelectedTime);
+    }
+    if (value === 'PM' && currentHours <= 12) {
+      setPeriod(value);
+      adjustedSelectedTime = setHours(currentHours + 12, adjustedSelectedTime);
+    }
+    updateCurrentTime(adjustedSelectedTime);
+  };
+
+  const updateCurrentTime = (time: Date) => {
+    setFocusedTime(time);
+    setSelectedTime(time);
+    onChange?.(time);
+  };
+
+  const onHourFocus = (hour: number) => {
+    const adjustedHour =
+      use12Hours && hour ? formatHourFrom12(hour, period) : hour;
+    applyFocus(adjustedHour, (value, date) => setHours(value, date));
```

Comment:
You can just pass `setHours` no need for arrow fn.

Reply:
Ok, `applyFocus` doesn't do anything, you just can call `setFocusTime` here.

Reply:
yup, simplified